cdc/sink/mysql.go

```diff
@@ -739,6 +773,26 @@ func (s *mysqlSink) execDMLs(ctx context.Context, rows []*model.RowChangedEvent,
 	return nil
 }
 
+// if the column value type is []byte and charset is not binary, we get its string
+// representation. Because if we use the byte array respresentation, the go-sql-driver
+// will automatically set `_binary` charset for that column, which is not expected.
+// See https://github.com/go-sql-driver/mysql/blob/master/connection.go#L267
+func appendQueryArgs(args []interface{}, col *model.Column) []interface{} {
+	var colValStr string
+	colValBytes, ok := col.Value.([]byte)
+	if ok {
+		colValStr = string(colValBytes)
+	}
+
+	if col.Charset != "" && col.Charset != charset.CharsetBin && ok {
+		args = append(args, colValStr)
```

Comment:
Could you move type assertion here? It's not cheap, we should avoid it as much as possible.